explore/future_compat.h

```diff
@@ -17,7 +17,11 @@
 #else
 #define VW_STD17_CONSTEXPR
 #define VW_ATTR(name)
-#    define VW_FALLTHROUGH  // fall through
+#    if defined(__GNUC__) && __GNUC__ >= 7
```

Comment:
do we need to update something here https://github.com/VowpalWabbit/vowpal_wabbit/blob/a9da12d32e7b4d027896762d98b3dcfdef34506e/explore/hash.h#L100

Reply:
Nope, it is used the same.